app/models/community_user.rb

```diff
@@ -77,11 +77,26 @@ def privilege(internal_id)
     UserAbility.joins(:ability).where(community_user_id: id, abilities: { internal_id: internal_id }).first
   end
 
-  def grant_privilege(internal_id)
+  ##
+  # Grant a specified ability to this CommunityUser.
+  # @param internal_id [String] The +internal_id+ of the ability to grant.
+  # @param notify [Boolean] Whether to send a notification to the user.
+  def grant_privilege!(internal_id, notify: true)
     priv = Ability.where(internal_id: internal_id).first
     UserAbility.create community_user_id: id, ability: priv
+    if notify
+      community_host = priv.community.host
+      user.create_notification("You've earned the #{priv.name} ability! Learn more.",
+                               url_helpers.ability_url(priv, host: community_host))
```

Comment:
`ability_url` needs to use the `internal_id` of the privilege - otherwise, clicking on the notification results in navigating to `/abilities/<numric_id>`, causing 404 (alternatively, what do you think about making this more robust by allowing navigation to the ability page by numeric id as well [i.e., `/abilities/2`]?).

Reply:
Huh, I read the route and saw `:id` and assumed numeric ID rather than `internal_id`... whoops.